LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/DefaultLoRaDataRequestHandler.cs

```diff
@@ -156,29 +177,28 @@ public async Task<LoRaDeviceRequestProcessResult> ProcessRequestAsync(LoRaReques
                         loRaDevice.UpdateRegion(request.Region.LoRaRegion, acceptChanges: false);
                 }
 
+                #region FrameCounterDown
                 // if deduplication already processed the next framecounter down, use that
                 var fcntDown = loRaADRResult?.FCntDown != null ? loRaADRResult.FCntDown : bundlerResult?.NextFCntDown;
 
-                if (fcntDown.HasValue)
-                {
-                    LogFrameCounterDownState(loRaDevice, fcntDown.Value);
-                }
-
-                // If it is confirmed it require us to update the frame counter down
+                // If we can send message downstream, we need to update the frame counter down
                 // Multiple gateways: in redis, otherwise in device twin
-                if (requiresConfirmation)
+                if (requiresConfirmation && !skipDownstreamToAvoidCollisions)
                 {
                     fcntDown = await EnsureHasFcntDownAsync(loRaDevice, fcntDown, payloadFcntAdjusted, frameCounterStrategy);
 
                     // Failed to update the fcnt down
-                    // In multi gateway scenarios it means the another gateway was faster than using, can stop now
+                    // In multi gateway scenarios it means another gateway has won the race to handle this message, can stop now
                     if (fcntDown <= 0)
                     {
                         return new LoRaDeviceRequestProcessResult(loRaDevice, request, LoRaDeviceRequestFailedReason.HandledByAnotherGateway);
                     }
                 }
+                #endregion
 
-                var validFcntUp = isFrameCounterFromNewlyStartedDevice || (payloadFcntAdjusted > loRaDevice.FCntUp);
+                var validFcntUp = isFrameCounterFromNewlyStartedDevice
```

Comment:
how about renaming this to canSendUpstream? It's not necessarily a valid fcnt in a sense that it incremented anymore, but we may still allow it to be sent upstream.

Reply:
6b50b16630e5fd0d7bf083b28bf9c6af12311829